tests/Aspire.MongoDB.Driver.Tests/ConformanceTests.cs

```diff
@@ -92,6 +104,11 @@ protected override void TriggerActivity(IMongoClient service)
         service.ListDatabases(source.Token);
     }
 
+    protected override HealthStatus GetHealthStatus()
+        => _containerFixture.Container?.Health == TestcontainersHealthStatus.Healthy
+            ? HealthStatus.Healthy
+            : HealthStatus.Unhealthy;
```

Comment:
We should always be expecting that we can connect to the server now and that it is healthy. We shouldn't need this new virtual/override method. We just need to `override bool CanConnectToServer => RequiresDockerTheoryAttribute.IsSupported;`, which will make the `HealthCheckReportsExpectedStatus` test work correctly.

Reply:
It was failing because the it got `Unhealthy` from the service.

https://github.com/radical/aspire/blob/46bce9b06b4e6ea10e3a68879bad53d24cb9c2a8/tests/Aspire.Components.Common.Tests/ConformanceTests.cs#L317-L320

Reply:
If the service says it is unhealthy, then the test should fail. We don't want to mask that problem. We want to know when something goes wrong.

Reply:
Right, and this would get the real status of the container to match against what we get from the health check. I'll try to see why/when it is showing unhealthy.